File: examples/contract_call/contract_call.rb
require_relative '../../wrappers/Ruby/aergo.rb'

aergo = Aergo.new("testnet-api.aergo.io", 7845)

account = AergoAPI::Account.new
# set the private key
account[:privkey] = "\xDB\x85\xDD\x0C\xBA\x47\x32\xA1\x1A\xEB\x3C\x7C\x48\x91\xFB\xD2\xFE\xC4\x5F\xC7\x2D\xB3\x3F\xB6\x1F\x31\xEB\x57\xE7\x24\x61\x76"
# or use an account on Ledger Nano S
#account[:use_ledger] = true
#account[:index] = 0

receipt = aergo.call_smart_contract(
    account,
    "AmgLnRaGFLyvCPCEMHYJHooufT1c1pENTRGeV78WNPTxwQ2RYUW7",
    "set_name",
    "Ruby!")

puts "status : " + receipt[:status]
puts "return : " + receipt[:ret]    
puts "BlockNo: " + receipt[:blockNo].to_s
puts "TxIndex: " + receipt[:txIndex].to_s
puts "GasUsed: " + receipt[:gasUsed].to_s
puts "feeUsed: " + receipt[:feeUsed].to_s

File: examples/contract_call/contract_call.cs
using System;

namespace contract_call_example
{
    class contract_call
    {
        static void Main(string[] args)
        {
            using (AergoClient client = new AergoClient())
            {
                client.Connect("testnet-api.aergo.io", 7845);

                AergoClient.AergoAccount account = new AergoClient.AergoAccount
                {
                    privkey = new byte[] {
                        0xDB, 0x85, 0xDD, 0x0C, 0xBA, 0x47, 0x32, 0xA1, 0x1A, 0xEB,
                        0x3C, 0x7C, 0x48, 0x91, 0xFB, 0xD2, 0xFE, 0xC4, 0x5F, 0xC7,
                        0x2D, 0xB3, 0x3F, 0xB6, 0x1F, 0x31, 0xEB, 0x57, 0xE7, 0x24,
                        0x61, 0x76
                    }
                };

                /* or use an account on Ledger Nano S */
                //AergoClient.AergoAccount account = new AergoClient.AergoAccount
                //{
                //    use_ledger = true
                //};

                Console.WriteLine("Calling Smart Contract...");

                AergoClient.TransactionReceipt receipt = client.CallSmartContract(ref account, "AmgLnRaGFLyvCPCEMHYJHooufT1c1pENTRGeV78WNPTxwQ2RYUW7", "set_name", "C# App");

                if (receipt.Sent) {
                    Console.WriteLine("--- Transaction Receipt ---");
                    Console.WriteLine("Status : " + receipt.status);
                    Console.WriteLine("ret    : " + receipt.ret);
                    Console.WriteLine("BlockNo: " + receipt.blockNo);
                    Console.WriteLine("TxIndex: " + receipt.txIndex);
                    Console.WriteLine("GasUsed: " + receipt.gasUsed);
                } else {
                    Console.WriteLine("Failed to send the transaction");
                }

                Console.WriteLine("Press any key to continue");
                Console.ReadKey();
            }
        }     
    }
}
